src/Material.cpp

```diff
@@ -1518,3 +1619,230 @@ Material* Material::clone(){
+ * @param number_density the number density of the Isotope in the Material
+ */  
+void IsoMaterial:: addIsotope(Isotope* isotope, FP_PRECISION number_density) {
+  int num_groups = isotope->getNumEnergyGroups();
+  bool chi_set = false;
+  
+  if (_num_groups==0) {
+    setNumEnergyGroups(num_groups);
+    _chi_set = false;
+  }
+  if (num_groups != _num_groups)
+    log_printf(ERROR,"Cannot add Isotope %d because it has %d energy groups "
+               "and other isotopes in Material have %d.", isotope->getId(),
+               num_groups, _num_groups);
+  
+  _isotopes.push_back(isotope);
+  _num_dens.push_back(number_density);
+  
+  for (int g=0; g<_num_groups; g++) {
+    _sigma_t[g] += isotope->getSigmaTByGroup(g+1)*number_density;
+    _sigma_a[g] += isotope->getSigmaAByGroup(g+1)*number_density;
+    _sigma_f[g] += isotope->getSigmaFByGroup(g+1)*number_density;
+    _nu_sigma_f[g] += isotope->getNuSigmaFByGroup(g+1)*number_density;
+    if (not _chi_set) {
+      _chi[g] = isotope->getChiByGroup(g+1);
+      if (_chi[g] > 0) chi_set = true;
+    }
+  }
+  
+  if (chi_set) _chi_set = true;
+  
+  if (isotope->isFissionable())
+    _fissionable = true;
+  
+  _num_isotopes++;
+}
+
+
+/**
+ * @brief Get specific number density from Material
+ * @param index The index of the material to be retrieved
+ * @return the number density
+ */
+FP_PRECISION IsoMaterial::getNumberDensity(int index) {
+  if (index > _num_isotopes-1)
+    log_printf(ERROR,"Cannot return Isotope with index %d because there are "
+               "only %d isotopes in the Material.", index,
+               _num_isotopes);
+
+  return _num_dens[index];
+}
+
+
+/**
+ * @brief Get vector of number densities from Material
+ * @return vector of number densities
+ */
+std::vector<FP_PRECISION> IsoMaterial::getNumberDensities() {
+  return _num_dens;
+}
+
+
+/**
+ * @brief Get specific Isotope from Material
+ * @param index The index of the material to be retrieved
+ * @return Pointer to the isotope requested
+ */
+Isotope* IsoMaterial::getIsotope(int index) {
+  if (index > _num_isotopes-1)
+    log_printf(ERROR,"Cannot return Isotope with index %d because there are "
+               "only %d isotopes in the Material.", index,
+               _num_isotopes);
+
+  return _isotopes[index];
+}
+
+
+/**
+ * @brief Get vector of Isotopes from Material
+ * @return vector of Isotopes
+ */
+std::vector<Isotope*> IsoMaterial::getIsotopes() {
+  return _isotopes;
+}
+  
+  
+  
+/**
+ * @brief Checks if the total cross-section for this Material is equal to the
+ *        absorption plus scattering cross-sections for all energy groups.
+ * @details If the total cross-section does not equal the absorption plus
+ *          scattering cross-section within SIGMA_T_THRESH then this method
+ *          exits OpenMOC.
+ */
+void IsoMaterial::checkSigmaT() {
+  if (_num_groups == 0)
+    log_printf(ERROR, "Unable to verify Material %d's total cross-section "
```

Comment:
Ditto my earlier comment - perhaps we want to make these ``WARNING`` rather than ``ERROR`` messages.